sound/soc/sof/ipc4-topology.c

```diff
@@ -2106,10 +2111,27 @@ static int sof_ipc4_route_setup(struct snd_sof_dev *sdev, struct snd_sof_route *
 	struct sof_ipc4_fw_module *sink_fw_module = sink_widget->module_info;
 	struct sof_ipc4_pipeline *src_pipeline = src_pipe_widget->private;
 	struct sof_ipc4_pipeline *sink_pipeline = sink_pipe_widget->private;
+	u32 core_id = src_pipeline->core_id;
 	struct sof_ipc4_msg msg = {{ 0 }};
 	u32 header, extension;
 	int ret;
 
+	if (sink_pipeline->core_id != core_id) {
+		dev_err(sdev->dev,
+			"pipelines must be on the same core: src: %u (%s), sink: %u (%s)\n",
+			core_id, src_pipe_widget->widget->name,
+			sink_pipeline->core_id, sink_pipe_widget->widget->name);
+		return -EINVAL;
+	}
```

Comment:
I don't think this rule is correct. It's perfectly fine to have a route between components that are handled by separate pipelines on separate cores. This would only allow for connected pipelines to be on the same core, and that's a short-term limitation we absolutely don't want to upstream.